src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1216,7 +1216,7 @@ static Image getShell32Icon(int iconID, int size) {
         long hIcon = getIconResource("shell32.dll", iconID, size, size);
         if (hIcon != 0) {
             Image icon = makeIcon(hIcon);
-            if (size != icon.getWidth(null)) {
+            if (icon != null && size != icon.getWidth(null)) {
                 icon = new MultiResolutionIconImage(size, icon);
```

Comment:
Although `icon != null` check is not strictly required due to `hIcon != 0` check, added it as sanity check and to be consistent with other locations in Win32ShellFolder2.java where makeIcon() is called (which is followed by a null check for the returned icon).

Reply:
I'd leave it out here. As @azuev-java [notes above](https://github.com/openjdk/jdk/pull/22776/files#r1887863199), the likely problem is `hIcon == 0` rather than `makeIcon` returns `null`.

Reply:
> > Although `icon != null` check is not strictly required here due to `hIcon != 0` check, added it as sanity check and to be consistent with other locations in Win32ShellFolder2.java where makeIcon() is called (which is followed by a null check for the returned icon).
> 
> I'd leave it out here. As @azuev-java [notes above](https://github.com/openjdk/jdk/pull/22776/files#r1887863199), the likely problem is `hIcon == 0` rather than `makeIcon` returns `null`.

Now that I've thought about it more, you're right `icon != null` is also needed… if the icon bits can't be extracted which can also occur if *GDI resources are exhausted*.

Reply:
Yes, I did [suggest it](https://github.com/openjdk/jdk/pull/22776#discussion_r1888597821) too.

On the other hand, `makeIcon` starts with this condition

https://github.com/openjdk/jdk/blob/4f44cf6bf2423a57a841be817f348e3b1e88f0eb/src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java#L1030-L1031

Anyway, not calling `makeIcon` when `hIcon == 0` is clearer.

Reply:
Do I retain or remove ` hIcon != 0L` check here ? Since now `hIcon != 0` is being checked before calling `mmakeIcon()`.

Reply:
> Do I retain or remove ` hIcon != 0L` check here ? Since now `hIcon != 0` is being checked before calling `mmakeIcon()`.

I'm for keeping it. Not calling a method is faster than calling it and returning immediately because a precondition isn't met. Plus, it makes it clear that we don't even try to create an icon from an invalid icon handle.